Classes/LoggerFactory.php

```diff
@@ -162,22 +116,35 @@ public static function getInstance()
     }
 
     /**
-     * @param string $identifier
-     * @param array $handlerConfiguration
-     * @return HandlerInterface
      * @throws InvalidConfigurationException
      */
-    protected function instanciateHandler($identifier, $handlerConfiguration)
+    protected function instanciateHandler(string $identifier, array $handlerConfiguration): HandlerInterface
     {
         if (!isset($this->handlerInstances[$identifier])) {
-            $handlerClass = isset($handlerConfiguration['className']) ? $handlerConfiguration['className'] : null;
+            $handlerClass = $handlerConfiguration['className'] ?? false;
 
             if (!class_exists($handlerClass)) {
                 throw new InvalidConfigurationException(sprintf('The given handler class "%s" does not exist, please check configuration for handler "%s".', $handlerClass, $identifier), 1436767219);
             }
 
             $arguments = (isset($handlerConfiguration['arguments']) && is_array($handlerConfiguration['arguments'])) ? $handlerConfiguration['arguments'] : [];
-            $this->handlerInstances[$identifier] = new $handlerClass(...$arguments);
+            $handler = new $handlerClass(...$arguments);
+
+            $formatterClass = $handlerConfiguration['formatterClassName'] ?? false;
+            if ($formatterClass) {
+                if (!class_exists($handlerClass)) {
```

Comment:
```suggestion
                if (!class_exists($formatterClass)) {
```

Reply:
And I guess this could merge with the previous if to if (!$formatterClass || !class_exists(.....
Neesd some more restructuring but could be nice maybe?

Reply:
If no formatter is given, that's totally fine…